### PR TITLE
Call security handler for ws endpoints

### PR DESCRIPTION
**Pull request by burbas**
Nova does not currently support plugins for websockets which makes it impossible to invoke the security-plugin in a dynamic way. So this commit hard-codes call for the security-handler when having a secure websocket path .

### CODE REVIEW

File: src/nova_ws_handler.erl [hunk 1]
```diff
@@ -40,8 +40,23 @@ init(Req, State = #{mod := Mod, controller_data := ControllerData}) ->
             ?ERROR("Websocket handler ~p returned unkown result ~p", [Mod, Error]),
             Req1 = cowboy_req:reply(500, Req),
             {ok, Req1, State}
+    end;
+init(Req, State) ->
+    %% We have something that needs securing
+    case nova_security_plugin:pre_http_request(State, #{}) of
```
Comment (Taure) [inline src/nova_ws_handler.erl:46]: In this case could we add a pre_ws_upgrade that we could have in our plugins? Then here we could apply pre_ws_plugins in correct prio order the same that we do for http request and for ws?

I think that would be a cleaner solution if you want to have more things before upgrading to a websocket.

File: src/nova_ws_handler.erl [hunk 2]
```diff
@@ -40,8 +40,23 @@ init(Req, State = #{mod := Mod, controller_data := ControllerData}) ->
...
+        {ok, State0 = #{req := Req}} ->
+            %% Recurse with secure set to false. This is hacky, but we can live with that now
+            init(Req, State0#{secure => false});
+        {stop, State0 = #{req := Req}} ->
+            %% We have no way to exit other than marking this is a unauthrized attempt and close in
+            %% websocket_init/1 function
+            init(Req, State0#{should_terminate => true,
```
Comment (Taure) [inline src/nova_ws_handler.erl:53]: Why don't we return 401 here?

File: src/nova_ws_handler.erl (hunk 1 above)
Comment (burbas) [reply src/nova_ws_handler.erl:46]: We already have support for a `pre_ws_request` so this should probably be this. Or a `pre_upgrade_ws_request`. I've already started on such a solution :)

File: src/nova_ws_handler.erl (hunk 2 above)
Comment (burbas) [reply src/nova_ws_handler.erl:53]: We can not do that according to the websocket behaviour from cowboy.

Comment (burbas) [reply src/nova_ws_handler.erl:53]: I was wrong. After digging around in cowboy source I found out that their documentation is partially wrong and the init-callback actually _can_ return a "normal" response. Will update this.

File: src/nova_ws_handler.erl (hunk 1 above)
Comment (Taure) [reply src/nova_ws_handler.erl]: yeah, meant pre_upgrade_ws_request.

**Pull request merged**